src/dali/abstract_ccxt_input_plugin.py

```diff
@@ -439,6 +439,16 @@ def _process_withdrawals(
             # End of pagination details
             pass
 
+    def _safe_api_call(
+        self,
+        function: Callable[..., Iterable[Dict[str, Union[str, float]]]],
+        params: Dict[str, Any],
+    ) -> Iterable[Dict[str, Union[str, float]]]:
+        return self.__safe_api_call(
+            function,
+            params
+        )
+
     def __safe_api_call(
         self,
         function: Callable[..., Iterable[Dict[str, Union[str, float]]]],
```

Comment:
Need to initialize Line 458 otherwise will get unbound exception

Reply:
You mean initialize `results` in Line 458? The exception is at Line 482? That shouldn't be happening. Can I see the stack trace?

Reply:
Unfortunately, I don't have it anymore. It was over the weekend. I think it was a case where `function(**params)` threw an exception and failed to assign results and the recovery logic reached its max limit for retries. To be fair, it was a bug in the way I was building the params variable and ccxt was complaining. I don't know if you count that as a bug since it was more of me learning how to use the kraken API. But it did confuse the hell out of me because I couldn't see what ccxt didn't like until tracking it down in their code.